### PR TITLE
Added model cleaned event

### PR DESCRIPTION
**Pull request by ghost**
Added cleaned event + tests

### CODE REVIEW

**Comment by ghost** [comment]
@freekmurze any idea why "Class 'App\Events\Event'" is not recognized?

**Comment by sebastiandedeyne** [comment]
I think the class is being removed in 5.3, either way it's not a good idea to extend a class that isn't part of Laravel's Illuminate library.

File: src/CleanUpModelsCommand.php [hunk 1]
```diff
@@ -64,6 +64,9 @@ protected function cleanUp(Collection $cleanableModels)
 
             $numberOfDeletedRecords = $class::cleanUp($class::query())->delete();
 
+            if($numberOfDeletedRecords > 0)
```
Comment (freekmurze) [inline src/CleanUpModelsCommand.php]: Please use curly braces for an if statement. Checking on `> 0` is not necessary in this case.

File: src/CleanUpModelsCommand.php [hunk 2]
```diff
@@ -110,4 +113,10 @@ protected function getFullyQualifiedClassNameFromFile(string $path) : string
             })
             ->first();
     }
+
+    protected function fireEvent(string $modelName, int $numberOfDeletedRecords){
+
+        event(new ModelCleanedEvent($modelName, $numberOfDeletedRecords));
```
Comment (freekmurze) [inline src/CleanUpModelsCommand.php]: You may drop this function and just use `event` inside the `cleanup` method.

File: src/ModelCleanedEvent.php [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Spatie\ModelCleanup;
+
+use App\Events\Event;
+use Illuminate\Queue\SerializesModels;
+
+class ModelCleanedEvent extends Event
+{
+
+    public $modelName;
```
Comment (freekmurze) [inline src/ModelCleanedEvent.php:14]: add docblocks here so it is clear which type of variable these properties hold

File: src/ModelCleanedEvent.php [hunk 4]
```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace Spatie\ModelCleanup;
+
+use App\Events\Event;
+use Illuminate\Queue\SerializesModels;
+
+class ModelCleanedEvent extends Event
```
Comment (freekmurze) [inline src/ModelCleanedEvent.php]: Extending `Event` is not necessary. Removing this will probably make the tests pass.

**Comment by freekmurze** [comment]
I added some comments to the code.
Could you also add a test to make sure that `$numberOfDeletedRecords` inside the event has the correct number?

**Comment by ghost** [comment]
Thank you for suggestions! I'll refactor the code as soon as possible!
@freekmurze Any idea of how can I access to $numberOfDeletedRecords of the event from within the test?
Sorry for this question, but it is the first time I've this doubt.
thanks

**Comment by freekmurze** [comment]
Have to figure that out myself too. I will add it as soon as I have a little time on my hands.

**Comment by freekmurze** [comment]
Thanks for contributing!

**Comment by freekmurze** [comment]
Version 1.1.0 has been tagged. I've added test that confirms the value of `numberOfDeletedRecords` is correct. Take a look in the source code.

**Pull request merged**